### DISCUSSION TITLE
Why is MediaPipe messing with my soundcard?

### DISCUSSION THREAD

**Discussion by mageofthesands** [Bing-su/adetailer#244] [Q&A]
This is weird, right?  After I click OK, the warning comes up again.  Clicking OK again lets Stable Diffusion continue.
As far as I can tell, when adetailer is using MediaPipe, it will do something with my sound card, and switch it back.  Today I did a clean install of SD DirectML.  I saw something about soundboard while adetailer was installing.  Now the ASIO popup happens when loading SD, requiring me to do the double click thing just to run SD.  
```
INFO: Created TensorFlow Lite XNNPACK delegate for CPU.[05:14<00:00, 12.04s/it]
mediapipe: 1 detected.
 15%|██████▌                                    | 2/13 [00:08<00:46,  4.24s/it]
```
![ASIO Mediapipe](https://github.com/Bing-su/adetailer/assets/81931767/38a79081-c104-4a44-a71c-7f37ed2b2fd5)

**Comment by Bing-su** [comment]
말씀하신 대로 mediapipe에는 오디오를 사용하는 기능이 포함되어있습니다. 이 문제의 해결법은 구글에 검색하면 다양하게 나오는 것 같으니, 사용자분의 환경에 맞는 해결법을 찾아서 실행해보시는게 좋겠습니다.

**Comment by TJYSunset** [comment]
Not sure if you still need a solution, but for those using MediaPipe and the only Google result is this page: [sounddevice](https://pypi.org/project/sounddevice/) scans system audio devices on load and causes the system audio interruption. You can prevent that module from loading with

```python
import sys
sys.modules["sounddevice"] = None
```

before importing mediapipe.

(Sorry, not exactly related to this project, and the hack probably shouldn't be committed to this project's repository; in the case of this project's users, prepending the hack to [`__init__.py`](adetailer/__init__.py) should presumably work.)
